internal/real_metric_registry.go

```diff
@@ -0,0 +1,234 @@
+package internal
+
+import (
+	"sync"
+	"time"
+)
+
+// metric registry for internal metrics
+type RealMetricRegistry struct {
+	source       string
+	prefix       string
+	tags         map[string]string
+	reportTicker *time.Ticker
+	sender       internalSender
+	done         chan struct{}
+
+	mtx     sync.Mutex
+	metrics map[string]interface{}
+
+	pointsValid   *DeltaCounter
+	pointsInvalid *DeltaCounter
+	pointsDropped *DeltaCounter
+
+	histogramsValid   *DeltaCounter
+	histogramsInvalid *DeltaCounter
+	histogramsDropped *DeltaCounter
+
+	spansValid   *DeltaCounter
+	spansInvalid *DeltaCounter
+	spansDropped *DeltaCounter
+
+	spanLogsValid   *DeltaCounter
+	spanLogsInvalid *DeltaCounter
+	spanLogsDropped *DeltaCounter
+
+	eventsValid   *DeltaCounter
+	eventsInvalid *DeltaCounter
+	eventsDropped *DeltaCounter
+}
+
+func NewMetricRegistry(sender internalSender, setters ...RegistryOption) *RealMetricRegistry {
```

Comment:
This should return the interface instead

Reply:
I made the change. I see that the compiler accepts either `*RealMetricRegistry` or the interface type `InternalSDKMetricRegistry`. Can you help me understand why you prefer the interface return type here?